From cda2802fe960737abb4623166ca4f75a63b5fe5a Mon Sep 17 00:00:00 2001
From: Thomas Haller <thaller@redhat.com>
Date: Thu, 8 Jan 2015 12:01:32 +0100
Subject: build: fix --enable-lto configure option to allow disabling option
 explicitly

When specifying '--enable-lto=anything' or '--disable-lto',
the configure script would always set enable_lto=yes.

The only way to disable lto, was *not* specifying the
configure option.

https://bugzilla.gnome.org/show_bug.cgi?id=742575
(cherry picked from commit 6eccfda0face594f86dea495d2ed2546b0d37e77)

diff --git a/configure.ac b/configure.ac
index b6bf38e..b407686 100644
--- a/configure.ac
+++ b/configure.ac
@@ -821,10 +821,11 @@ AM_CONDITIONAL(BUILD_NMTUI, test "$build_nmtui" = yes)
 
 NM_COMPILER_WARNINGS
 
-AC_ARG_ENABLE(lto, AS_HELP_STRING([--enable-lto], [Enable Link Time Optimization for smaller size (default: no)]),
-              [enable_lto=yes], [enable_lto=no])
+AC_ARG_ENABLE(lto, AS_HELP_STRING([--enable-lto], [Enable Link Time Optimization for smaller size (default: no)]))
 if (test "${enable_lto}" = "yes"); then
     CFLAGS="-flto $CFLAGS"
+else
+    enable_lto='no'
 fi
 
 
-- 
cgit v0.10.2

